src/commands/basic/genbanlist.js

```diff
@@ -0,0 +1,55 @@
+const fetch = require("node-fetch")
+const { apiurl } = require("../../../config.json")
+const ConfigModel = require("../../database/schemas/config")
+const { MessageEmbed } = require("discord.js");
+const Str = require("@supercharge/strings")
+const fs = require("fs")
+
+module.exports = {
+    config: {
+        name: "genbanlist",
+        aliases: [],
+        usage: "",
+        category: "basic",
+        description: "Creates a .json banlist file to use for servers",
+        accessibility: "Moderator",
+    },
+    run: async (client, message, args) => {
+        const config = await ConfigModel.findOne({guildid: message.guild.id})
+        if (!config.trustedCommunities) return message.reply("Please set trusted communities first")
+        if (!config.ruleFilters) return message.reply("Please set rule filters first")
+        message.reply("Processing banlist. Please wait")
+
+        let trustedCommunitiesProm = config.trustedCommunities.map(id => {
+            return fetch(`${apiurl}/communities/getid?id=${id}`).then(res => res.text())
+        })
+        const trustedCommunities = await Promise.all(trustedCommunitiesProm)
+        // console.log(apiurl)
+
+        let rulePromises = config.ruleFilters.map((rule) => {
+            return fetch(`${apiurl}/violations/getbyrule?id=${rule}`).then(res => res.json())
+        })
+        let ruleViolations = await Promise.all(rulePromises)
+        let nameArr = []
+        ruleViolations.forEach((violations) => {
+            if (violations === []) return
+            violations.forEach((violation => {
+                nameArr.push(violation.playername)
+            }))
+        })
+        nameArr = nameArr.filter((name, i) => nameArr.indexOf(name) === i)
+        let banlist = nameArr.map((name) => {
+            return {
+                username: name,
+                reason: "Banned on FAGC. Please check one of the community Discord servers"
```

Comment:
I added in the violation by ID, but the issue is, the player can be added to the banlist only once, even if they have hundreds of thousands of violations, so only the first one that was fetched would be linked, which might not be ideal.

Reply:
Another issue is that currently, you can't click links in factorio. so something like http://localhost:3000/v1/violations/getbyid?id=6068a92d6ac5a8f0dde1722b would need to be manually typed out by players. Not sure about convenience, but it's there

Reply:
I was thinking more along the lines of `${apiurl}/offenses/getall?playername=${playername}` so the player can find out where they have offenses.

Reply:
Yup, that is much better to type out. It will however show irrelevant offenses and violations too, such as if the community doesn't ban for LHD but the player is banned for LHD.

Reply:
I think it is ok for it to have irrelevant ones as the alternative is a url people might never visit.